contracts/governance/liquidation/LiquidationStrategy.sol

```diff
@@ -0,0 +1,170 @@
+import {ILiquidationStrategy} from '../../interfaces/ILiquidationStrategy.sol';
+import {ILiquidationCallback} from '../../interfaces/ILiquidationCallback.sol';
+import {ITreasuryPool} from '../../interfaces/ITreasuryPool.sol';
+
+contract LiquidationStrategy is ILiquidationStrategy, PermissionAdmin, Utils, ReentrancyGuard {
+
+  using SafeERC20 for IERC20Ext;
+  using SafeMath for uint256;
+
+  // after repeatedPeriod since startTime, there will be duration (in seconds)
+  // for liquidators to liquidate tokens in the fee pool
+  // for example: from deployed time, every 2 weeks liquidation is enabled for 4 days
+  struct LiquidationSchedule {
+    uint128 startTime;
+    uint64 repeatedPeriod;
+    uint64 duration;
+  }
+
+  LiquidationSchedule public liquidationSchedule;
+  ITreasuryPool public feePool;
+  address payable public treasuryPool;
+  // list of tokens that can be liquidate to
+  mapping(address => bool) public whitelistedTokens;
+
+  event FeePoolSet(address indexed feePool);
+  event TreasuryPoolSet(address indexed treasuryPool);
+  event LiquidationScheduleUpdated(uint128 startTime, uint64 repeatedPeriod, uint64 duration);
+  event WhitelistedTokenUpdated(address indexed token, bool indexed isAdd);
+
+  modifier onlyWhenLiquidationEnabled() {
+    require(isLiquidationEnabled(), 'only when liquidation enabled');
+    _;
+  }
+
+  constructor(
+    address _admin,
+    address _feePool,
+    address payable _treasuryPool,
+    uint128 _startTime,
+    uint64 _repeatedPeriod,
+    uint64 _duration,
+    address[] memory _whitelistedTokens
+  ) PermissionAdmin(_admin) {
+    _setFeePool(_feePool);
+    _setTreasuryPool(_treasuryPool);
+    _setLiquidationSchedule(_startTime, _repeatedPeriod, _duration);
+    _updateWhitelistedToken(_whitelistedTokens, true);
+  }
+
+  function updateLiquidationSchedule(
+    uint128 _startTime,
+    uint64 _repeatedPeriod,
+    uint64 _duration
+  )
+    external onlyAdmin
+  {
+    _setLiquidationSchedule(_startTime, _repeatedPeriod, _duration);
+  }
+
+  function updateFeePool(address _feePool) external onlyAdmin {
+    _setFeePool(_feePool);
+  }
+
+  function updateTreasuryPool(address payable _treasuryPool) external onlyAdmin {
+    _setTreasuryPool(_treasuryPool);
+  }
+
+  function updateWhitelistedTokens(address[] calldata _tokens, bool _isAdd)
+    external override onlyAdmin
+  {
+    _updateWhitelistedToken(_tokens, _isAdd);
+  }
+
+  /** @dev Liquidate list of tokens to a single dest token,
+  *   source token must not be a whitelisted token, dest must be a whitelisted token
+  * @param sources list of source tokens to liquidate
+  * @param amounts list of amounts corresponding to each source token
+  * @param recipient receiver of source tokens
+  * @param dest token to liquidate to, must be whitelisted
+  * @param minReturn minimum return of dest token for this liquidation
+  * @param txData data to callback to recipient
+  */
+  function liquidate(
```

Comment:
this should be `internal`, E.g operator should not fill in minReturn.

Reply:
Updated to internal